### PR TITLE
Add support for ProgID registration

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This only represents the registration part of the work. There is still work to be done in the SDK to generate a `.clsidmap` file containing a ProgID.

cc @jkoritzinsky 

See: https://github.com/dotnet/coreclr/issues/25966

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
I have validated this with a custom generated clsidmap instance and it behaves as expected.

**Comment by AaronRobinsonMSFT** [comment]
This will need to be ported to release/3.0 as well.

cc @MeiChin-Tsai @jeffschwMSFT

**Comment by elinor-fung** [comment]
Should we add a test that uses this (at least in master)?

File: src/corehost/cli/comhost/comhost.cpp [hunk 1]
```diff
@@ -309,6 +406,43 @@ namespace
                 return __HRESULT_FROM_WIN32(res);
         }
 
+        // Check if a Prog ID is defined
+        if (!entry.progid.empty())
+        {
+            // Register the ProgID in the CLSID key
+            WCHAR regKeyProgIdPath[ARRAYSIZE(regKeyClsidPath) * 2];
+            ::swprintf_s(regKeyProgIdPath, L"%s\\ProgId", regKeyClsidPath);
```
Comment (elinor-fung) [inline src/corehost/cli/comhost/comhost.cpp]: nit: I know it doesn't actually matter, but all the docs case it `ProgID`

**Comment by AaronRobinsonMSFT** [comment]
> Should we add a test that uses this (at least in master)?

I will do that in a future PR. Right now this work is purely a no-op since it is entirely related to the COM system consuming the mapping. I have updated https://github.com/dotnet/core-setup/issues/5113.

**Pull request merged**